Sources/OpenAPIRuntime/Interface/Body.swift

```diff
@@ -0,0 +1,535 @@
+    public let length: Length
+
+    /// The underlying type-erased async sequence.
+    private let sequence: BodySequence
+
+    /// A lock for shared mutable state.
+    private let lock: NSLock = {
+        let lock = NSLock()
+        lock.name = "com.apple.swift-openapi-generator.runtime.body"
+        return lock
+    }()
+
+    /// Whether an iterator has already been created.
+    private var locked_iteratorCreated: Bool = false
+
+    private init(
+        sequence: BodySequence,
+        length: Length,
+        iterationBehavior: IterationBehavior
+    ) {
+        self.sequence = sequence
+        self.length = length
+        self.iterationBehavior = iterationBehavior
+    }
+}
+
+extension Body: Equatable {
+    public static func == (
+        lhs: Body,
+        rhs: Body
+    ) -> Bool {
+        ObjectIdentifier(lhs) == ObjectIdentifier(rhs)
+    }
+}
+
+extension Body: Hashable {
+    public func hash(into hasher: inout Hasher) {
+        hasher.combine(ObjectIdentifier(self))
+    }
+}
+
+// MARK: - Creating the Body
+
+extension Body {
+
+    public convenience init(
+        data: DataType,
+        length: Length
+    ) {
+        self.init(
+            dataChunks: [data],
+            length: length
+        )
+    }
+
+    public convenience init(
+        data: DataType
+    ) {
+        self.init(
+            dataChunks: [data],
+            length: .known(data.count)
+        )
+    }
+
+    public convenience init<S: Sequence>(
+        dataChunks: S,
+        length: Length,
+        iterationBehavior: IterationBehavior
+    ) where S.Element == DataType {
+        self.init(
+            sequence: .init(WrappedSyncSequence(sequence: dataChunks)),
+            length: length,
+            iterationBehavior: iterationBehavior
+        )
+    }
+
+    public convenience init<C: Collection>(
+        dataChunks: C,
+        length: Length
+    ) where C.Element == DataType {
+        self.init(
+            sequence: .init(WrappedSyncSequence(sequence: dataChunks)),
+            length: length,
+            iterationBehavior: .multiple
+        )
+    }
+
+    public convenience init<C: Collection>(
+        dataChunks: C
+    ) where C.Element == DataType {
+        self.init(
+            sequence: .init(WrappedSyncSequence(sequence: dataChunks)),
+            length: .known(dataChunks.map(\.count).reduce(0, +)),
+            iterationBehavior: .multiple
+        )
+    }
+
+    public convenience init(
+        stream: AsyncThrowingStream<DataType, any Error>,
+        length: Body.Length
+    ) {
+        self.init(
+            sequence: .init(stream),
+            length: length,
+            iterationBehavior: .single
+        )
+    }
+
+    public convenience init(
+        stream: AsyncStream<DataType>,
+        length: Body.Length
+    ) {
+        self.init(
+            sequence: .init(stream),
+            length: length,
+            iterationBehavior: .single
+        )
+    }
+
```

Comment:
These two methods seem unnecessary when you already have the generic one which takes an `AsyncSequence`

Reply:
It's subtle, but we have them here because we know that async streams are _never_ safe to iterate multiple times (so we hardcode `.single` here, whereas a generic AsyncSequence might be, depends on the sequence and that's why those initializers force the user to specify either way.